src/filesystem/drive/access.rs

```diff
@@ -301,7 +303,7 @@ impl DriveAccess {
             return Err(DriveAccessError::ActorAlreadyPresent);
         }
 
-        let mut actor_settings = ActorSettings::new(key, access_mask);
+        let mut actor_settings = ActorSettings::new(key, access_mask, actor_id);
```

Comment:
Our `ActorId` is derivable from the key so I don't believe it needs to get passed in here.